crates/uv-resolver/src/pubgrub/priority.rs

```diff
@@ -88,6 +102,66 @@ impl PubGrubPriorities {
             PubGrubPackageInner::Package { name, .. } => self.0.get(name).copied(),
         }
     }
+
+    /// Returns whether the priority was changed, i.e., it's the first time we hit this condition
+    /// for the package.
+    pub(crate) fn make_conflict_early(&mut self, package: &PubGrubPackage) -> bool {
+        let next = self.0.len();
+        let Some(name) = package.name_no_root() else {
+            // Not a correctness bug
+            assert!(
+                !cfg!(debug_assertions),
+                "URL packages must not be involved in conflict handling"
+            );
```

Comment:
Personally, I'd do this:

```rust
#[cfg(debug_assertions)]
{
    panic!("yadda yadda yadda");
}
```

It took me a second to read what this assert was doing, but I think the above is a little easier to read.